frontend/src/app/components/containers/pages/verify-access.tsx

```diff
@@ -0,0 +1,156 @@
+import { FC, useEffect, useMemo, useState } from "react";
+import { Link, useLocation } from "react-router-dom";
+import logoLg from "../../../../assets/images/branding/BCgov-lg.png";
+import { Footer } from "../layout";
+import { TbFaceId, TbFaceIdError, TbLockAccess } from "react-icons/tb";
+import { validate as uuidValidate } from "uuid";
+import { useAppDispatch, useAppSelector } from "../../../hooks/hooks";
+import { validateComsAccess } from "../../../store/reducers/app";
+import config from "../../../../config";
+import EnvironmentBanner from "../layout/environment-banner";
+import { IconType } from "react-icons/lib";
+
+type Props = {};
+
+const useQuery = () => {
+  const { search } = useLocation();
+  return useMemo(() => new URLSearchParams(search), [search]);
+};
+
+export const VerifyAccess: FC<Props> = () => {
+  const dispatch = useAppDispatch();
+  const query = useQuery();
+
+  const [isValid, setIsValid] = useState<boolean | undefined>();
+  const [wasRun, setWasRun] = useState(false);
+
+  const environmentName = config.ENVIRONMENT_NAME || "production";
+  const isLoading = useAppSelector((state) => state?.app?.loading?.isLoading);
+
+  useEffect(() => {
+    const token = query.get("token");
+
+    if (!isLoading && !wasRun) {
+      if (token && uuidValidate(token) && !wasRun) {
```

Comment:
nitpik: I think the !wasRun is redundant here (since we can only get here if !wasRun is true above).

Reply:
There's some method to that madness. isLoading is waiting for all network requests to finish, and wasRun is tracking the verification. Removing !wasRun will cause an infinite render, and in its current position its causing the infinite render on bad or reused token